pytorch_lightning/accelerators/accelerator.py

```diff
@@ -0,0 +1,365 @@
+from pytorch_lightning.utilities.enums import LightningEnum
+from typing import Any, Callable, Iterable, Optional, Union
+
+import torch
+from torch.optim import Optimizer
+
+from pytorch_lightning.plugins import TrainingTypePlugin, HorovodPlugin
+from pytorch_lightning.plugins.precision import (
+    ApexMixedPrecisionPlugin,
+    MixedPrecisionPlugin,
+    NativeMixedPrecisionPlugin,
+    PrecisionPlugin,
+)
+from pytorch_lightning.core import LightningModule
+from pytorch_lightning.utilities import AMPType
+from pytorch_lightning.utilities.apply_func import move_data_to_device
+
+
+class Accelerator(object):
+    """
+    The Accelerator Base Class.
+    An Accelerator is meant to deal with one type of Hardware.
+
+    Currently there are accelerators for:
+    - CPU
+    - GPU
```

Comment:
@SeanNaren if we want to do ZeRO-v3 and offload more to CPU memory, is that handled within the training type plugin entirely? is there any dependency on the accelerator in which the plugin is contained?

Reply:
yeah handled explicitly within the training type plugin, or more technically, within the sharded DDP/OSS class itself. I think from what I recall, an `offload_device` will be exposed giving the user control (in this case, our accelerator control)